code/rounds/InProgressRound.cs

```diff
@@ -175,11 +161,11 @@ private TTTTeam IsRoundOver()
         private void AssignRoles()
         {
             // TODO: There should be a neater way to handle this logic.
-            int traitorCount = (int) Math.Max(Players.Count * 0.25f, 1f);
+            int traitorCount = (int) Math.Max(_startingRoundPlayers.Count * 0.25f, 1f);
 
             for (int i = 0; i < traitorCount; i++)
             {
-                List<TTTPlayer> unassignedPlayers = Players.Where(p => p.Role is NoneRole).ToList();
+                List<TTTPlayer> unassignedPlayers = _startingRoundPlayers.Where(p => p.Role is NoneRole).ToList();
```

Comment:
make sure, these players still exists (could have disconnected meanwhile)

Reply:
Let me know if what I have now is fine.